File: 1850-minimum-length-of-string-after-deleting-similar-ends/minimum-length-of-string-after-deleting-similar-ends.js
/**
 * @param {string} s
 * @return {number}
 */
var minimumLength = function(s) {
    let l = 0, r = s.length - 1;
    while (l < r && s[l] === s[r]) {
        let ch = s[l];
        while (l <= r && s[l] === ch) l++;
        while (l <= r && s[r] === ch) r--;
    }
    return  r - l + 1;
};

File: 1850-minimum-length-of-string-after-deleting-similar-ends/minimum-length-of-string-after-deleting-similar-ends.cs
public class Solution {
    public int MinimumLength(string s) {
        int l = 0, r = s.Length - 1;
        while (l < r && s[l] == s[r]) {
            char ch = s[l];
            while (l <= r && s[l] == ch) l++;
            while (l <= r && s[r] == ch) r--;
        }
        return r - l + 1;
    }
}